Objects/memoryobject.c

```diff
@@ -3172,17 +3172,65 @@ typedef struct {
 } memoryiterobject;
 
 
+static void memoryiter_dealloc(memoryiterobject *);
+static int memoryiter_traverse(memoryiterobject * , visitproc, void *);
+static PyObject * memoryiter_next(memoryiterobject *);
+static PyObject *memory_iter(PyObject *);
+
+
 static PyTypeObject PyMemoryIter_Type = {
     PyVarObject_HEAD_INIT(&PyType_Type, 0)
     .tp_name = "memory_iterator",
     .tp_basicsize = sizeof(memoryiterobject),
     // methods
+    .tp_dealloc = (destructor)memoryiter_dealloc,
     .tp_getattro = PyObject_GenericGetAttr,
     .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
+    .tp_traverse = (traverseproc)memoryiter_traverse,
     .tp_iter = PyObject_SelfIter,
+    .tp_iternext = (iternextfunc)memoryiter_next,
 };
 
 
+static void
+memoryiter_dealloc(memoryiterobject *it)
+{
+  _PyObject_GC_UNTRACK(it);
+  Py_XDECREF(it->it_seq);
+  PyObject_GC_Del(it);
+}
+
+static int
+memoryiter_traverse(memoryiterobject *it, visitproc visit, void *arg)
+{
+  Py_VISIT(it->it_seq);
+  return 0;
+}
+
+static PyObject *
+memoryiter_next(memoryiterobject *it)
+{
+  PyMemoryViewObject *seq;
+
+  assert(it != NULL);
+  seq = it->it_seq;
+  if (seq == NULL)
+      return NULL;
+  assert (PyMemoryView_Check(seq));
+
+  if (it->it_index < PyMemoryView_GET_SIZE(seq)){
+      PyObject * item = PyLong_FromSsize_t(seq->view.itemsize);
```

Comment:
Okay, this returns the item size each time. You know what that is, right? It's the size (in bytes) of the items. If you use `memoryview(b"...")` this is always 1.

The next step would be to use `PyObject_GetItem()` to get the Nth item (N == it_index).

Reply:
> You know what that is, right? 
Yes, I was just trying to get it to compile as well as push the update with the complimentary functions. (and looking for a "get_item" function)

I will continue working on it today.